CtfPlayback/Metadata/AntlrParser/CtfAntlrMetadataParser.cs

```diff
@@ -110,6 +110,16 @@ private unsafe string GetMetadata(Stream metadataStream)
                         break;
                     }
 
+                    if (read != packetSize) // .NET 6.0 breaking change - https://learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/6.0/partial-byte-reads-in-streams - bytes read can be less than what was requested
+                    {
+                        while (read < buffer.Length) // Keep reading until we fill up our buffer to the packetSize
+                        {
+                            int tmpBytesRead = metadataStream.Read(buffer.AsSpan().Slice(read));
+                            if (tmpBytesRead == 0) break;
+                            read += tmpBytesRead;
+                        }
+                    }
+
```

Comment:
we might want a check that "read" >= contentSize. I'm thinking this might be a corrupt file otherwise?

Reply:
Not sure this is a good check. Read reads up until packetSize which can be less than contentSize